appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/util/ModelUtils.java

```diff
@@ -313,12 +313,6 @@ public static SchemaType getParentSchemaType(SchemaType type1, SchemaType type2)
         if (type2 == null) {
             return type1;
         }
-        if (type1 == SchemaType.OBJECT || type2 == SchemaType.OBJECT) {
-            return SchemaType.OBJECT;
-        }
-        if (type1 == SchemaType.STRING || type2 == SchemaType.STRING) {
-            return SchemaType.STRING;
-        }
```

Comment:
FYI: the above 2 cases aren't required as the default `return type1` (below, not visible) will effectively do the same while the other check (`type1 != type2`) done will not match.

Reply:
Is this correct? As I see it the results have changed:

| type1  	| type2  	| old parent 	| new parent 	|
|--------	|--------	|------------	|------------	|
| OBJECT 	| STRING 	| OBJECT     	| STRING     	|
| OBJECT 	| NUMBER 	| OBJECT     	| STRING     	|

Reply:
I think it becomes clear if you look at the fill method:
```java
if (type1 == null && type2 == null) {
            return null;
        }
        if (type1 == null) {
            return type2;
        }
        if (type2 == null) {
            return type1;
        }
        // here we removed the logic that would return OBJECT of both are OBJECT or STRING if both are STRING
        if (type1 != type2) {
            return SchemaType.STRING;
        }
        return type1;
}
```
So the removed logic returns the common type of both are the same. If we remove it it now falls through, first tested if the are not the same (which for same has no effect) what brings us to do `return type1` (which for any same pair is the common type.